lib/ace/edit_session.js

```diff
@@ -324,6 +324,19 @@ var EditSession = function(text, mode) {
     };
 
     /**
+     * @param {Number} row
+     * 
+     * Returns context at row.
+     **/
+    this.getContext = function(row) {
```

Comment:
it will remain useful even if moved outside of the editor code. The issue is that it relies on particular convention about the state type (it may be string but it might be array as well) and semantics. It is our own convention and it is natural for our own conventions to be in our own code.